Model/Queue.php

```diff
@@ -470,28 +473,30 @@ protected function fetchJobs($jobsLimit, $fetchFullReindexJobs = false, $lastJob
             $rawJobsCount = count($rawJobs);
 
             $offset += $limit;
-            $limit = max(0, $maxJobs - $rawJobsCount);
+            $limit = max(0, $jobsLimit - $rawJobsCount);
 
             // $jobs will always be completely set from $rawJobs
             // Without resetting not-merged jobs would be stacked
             $jobs = [];
 
-            if (count($rawJobs) === $maxJobs) {
+            // At this point, if this condition is true, this means that no merge was possible in the last iteration
+            if (count($rawJobs) === $jobsLimit) {
                 $jobs = $rawJobs;
 
                 break;
             }
 
+            // Introduced an array of job sizes to determine the total batch size currently processed (sum of all jobs contained in the run)
+            // This will determine if we can continue to loop over the jobs
+            $jobSizes = [];
```

Comment:
Not sure why the earlier accumulation was failing but this seems more straightforward to understand and I like that we're getting rid of `break 2` 😅